src/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -50,15 +50,74 @@ public enum GCNotificationStatus
         NotApplicable = 4
     }
 
+    public struct GCMemoryInfo
+    {
+        /// <summary>
+        /// HighMemoryLoadThreshold when the last GC occured
+        /// </summary>
+        public int HighMemoryLoadThreshold { get; }
+
+        /// <summary>
+        /// Total available memory for the GC to use when the last GC ocurred. By default this is the physical memory on the machine, but it may be customized by specifying a HardLimit.
+        /// </summary>
+        public long TotalAvailableMemory { get; }
+
+        /// <summary>
+        /// Memory Load when the last GC ocurred
+        /// </summary>
+        public int MemoryLoad { get; }
+
+        /// <summary>
+        /// The total heap size when the last GC ocurred
+        /// </summary>
+        public IntPtr HeapSize { get; }
```

Comment:
Why not to expose these as `long` as well? It is odd that `TotalAvailableMemory` is long, but these are `long`.

Reply:
on 32-bit systems, TotalAvailableMemory can be more than 32-bit but heap size will not be.

Reply:
The problem is that you cannot do any math with `IntPtr`. There are many places in framework that can technically be IntPtrs or UIntPtr, but that are `long` instead for usability. For example, `Environment.WorkingSet`.

Reply:
ahh, makes sense. I'm fine with making it a `long`.

Reply:
Ok, will make it long then.